src/react/hooks/juicebox/useProjectDistributionLimit.ts

```diff
@@ -0,0 +1,37 @@
+import { useContext } from "react";
+import { JBCurrency } from "../../../constants";
+import { useJbFundAccessConstraintsStoreDistributionLimitOf } from "../../generated/hooks";
+import { JBContractContext, useJBContractContext } from "src/react/contexts";
+
+/**
+ * The distribution limit for the project. The maximum amount of funds that can be distributed from the project in a funding cycle.
+ */
+export type DistributionLimit = {
+  /**
+   * The amount of the distribution limit in wei.
+   */
+  amountWei: bigint | undefined,
+  /**
+   * The currency of the distribution limit.
+   */
+  currency: JBCurrency | undefined,
+};
+
+/**
+ * Hook to retrieve the distribution limit for the current project.
+ *
+ * @returns The distribution limit for the current project.
+ */
+export function useProjectDistributionLimit(): DistributionLimit {
+  const { contracts: { fundAccessConstraintsStore } } = useJBContractContext();
+  const distributionLimit = useJbFundAccessConstraintsStoreDistributionLimitOf({
+    address: fundAccessConstraintsStore.data,
+  });
+
+  const currency = distributionLimit?.data?.[1]
+
+  return {
+    amountWei: distributionLimit?.data?.[0],
```

Comment:
**suggestion** return `new Ether(distributionLimit.data[0])` here if the value exists would be nice.

I would actually return a structure like this, so we retain the query data:

```jsx
return {
   data: {
       amountWei: ...
       currency: ...
   },
...distributionLimit
}
```

Reply:
Done